crates/forge/src/cmd/clone.rs

```diff
@@ -618,13 +667,240 @@ impl EtherscanClient for Client {
     }
 }
 
+/// SourcifyClient is a client for interacting with Sourcify API.
+pub(crate) struct SourcifyClient {
+    client: reqwest::Client,
+    chain: Chain,
+    base_url: String,
+}
+
+impl SourcifyClient {
+    pub fn new(chain: Chain) -> Self {
+        Self {
+            client: reqwest::Client::new(),
+            chain,
+            base_url: "https://sourcify.dev/server".to_string(),
+        }
+    }
+
+    fn get_contract_url(&self, address: Address, fields: &str) -> String {
+        format!("{}/v2/contract/{}/{}?fields={}", self.base_url, self.chain.id(), address, fields)
+    }
+}
+
+/// Sourcify API response for contract files.
+#[derive(Debug, Clone, Deserialize)]
+#[serde(untagged)]
+enum SourcifyContractResponse {
+    Success(Box<SourcifyContractData>),
+    Error(SourcifyErrorResponse),
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct SourcifyContractData {
+    #[serde(default)]
+    files: Option<BTreeMap<String, String>>,
+    #[serde(default)]
+    abi: Option<Box<serde_json::value::RawValue>>,
+    #[serde(default)]
+    compilation: Option<SourcifyCompilation>,
+    #[serde(default)]
+    #[allow(dead_code)]
+    creation_code: Option<String>,
+    #[serde(default)]
+    #[allow(dead_code)]
+    deployed_bytecode: Option<String>,
+    #[serde(default)]
+    #[allow(dead_code)]
+    runtime_bytecode: Option<String>,
+    #[serde(default)]
+    creation_transaction_hash: Option<String>,
+    #[serde(default)]
+    creator_address: Option<String>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct SourcifyCompilation {
+    #[serde(default)]
+    compiler_version: String,
+    #[serde(default)]
+    name: String,
+    #[serde(default)]
+    settings: Option<serde_json::Value>,
```

Comment:
this should be `compiler_settings` see https://sourcify.dev/server/v2/contract/1/0xDb53f47aC61FE54F456A4eb3E09832D08Dd7BEec?fields=sources,abi,compilation for a full response example